src/glib/base/dt.h

```diff
@@ -175,44 +176,63 @@ class TMem;
 typedef TPt<TMem> PMem;
 
 /// Memory chunk - advanced memory buffer
-class TMem : public TMemBase {
+class TMem {
 private:
   TCRef CRef;
 public:
   friend class TPt<TMem>;
 protected:
+  int MxBfL, BfL;
+  char* Bf;
   void Resize(const int& _MxBfL);
   bool DoFitLen(const int& LBfL) const {return BfL+LBfL<=MxBfL;}
 public:
-  TMem(const int& _MxBfL=0) : TMemBase() {
-    IAssert(BfL >= 0); MxBfL = _MxBfL; BfL = 0; Bf = NULL; Owner = true;
+  TMem(const int& _MxBfL=0) : MxBfL(_MxBfL), BfL(0), Bf(NULL) {
+    IAssert(BfL >= 0); BfL = 0; Bf = NULL;
     if (MxBfL>0){Bf=new char[MxBfL]; IAssert(Bf!=NULL);}}
   static PMem New(const int& MxBfL=0){return new TMem(MxBfL);}
-  TMem(const void* _Bf, const int& _BfL) : TMemBase() {
-    IAssert(BfL >= 0); MxBfL = _BfL; BfL = _BfL; Bf = NULL; Owner = true;
+  TMem(const void* _Bf, const int& _BfL) : MxBfL(_BfL), BfL(_BfL), Bf(NULL) {
+    IAssert(BfL >= 0); MxBfL = _BfL; BfL = _BfL; Bf = NULL;
     if (BfL > 0) { Bf = new char[BfL]; IAssert(Bf != NULL); memcpy(Bf, _Bf, BfL); } }
   static PMem New(const void* Bf, const int& BfL){return new TMem(Bf, BfL);}
-  TMem(const TMem& Mem) : TMemBase() {
-    MxBfL = Mem.MxBfL; BfL = Mem.BfL; Bf = NULL; Owner = true;
+  TMem(const TMem& Mem) : MxBfL(0), BfL(0), Bf(NULL) {
+    MxBfL = Mem.MxBfL; BfL = Mem.BfL; Bf = NULL;
     if (MxBfL>0){Bf=new char[MxBfL]; memcpy(Bf, Mem.Bf, BfL);}}
   static PMem New(const TMem& Mem){return new TMem(Mem);}
   static PMem New(const PMem& Mem){return new TMem(*Mem);}
   TMem(const TStr& Str);
-  TMem(TMem&& Src) : TMemBase() {
-    MxBfL = Src.MxBfL; BfL = Src.BfL; Bf = Src.Bf; Owner = Src.Owner;
-    Src.MxBfL = Src.BfL = 0; Src.Bf = NULL;  Src.Owner = false;
+  TMem(TMem&& Src) : MxBfL(0), BfL(0), Bf(NULL) {
+    MxBfL = Src.MxBfL; BfL = Src.BfL; Bf = Src.Bf;
+    Src.MxBfL = Src.BfL = 0; Src.Bf = NULL;
   }
   static PMem New(const TStr& Str){return new TMem(Str);}
-  ~TMem() { if (Owner && Bf != NULL) { delete[] Bf; }; Owner = false; Bf = NULL; }
+  ~TMem() { if (Bf != NULL) { delete[] Bf; }; Bf = NULL; }
   explicit TMem(TSIn& SIn) {
     SIn.Load(MxBfL); SIn.Load(BfL);
-    Bf = new char[MxBfL = BfL]; SIn.LoadBf(Bf, BfL); Owner = true; }
+    Bf = new char[MxBfL = BfL]; SIn.LoadBf(Bf, BfL); }
+
+  int Len() const { return BfL; }
+  bool Empty() const { return BfL == 0; }
+  char* GetBf() const { return Bf; }
+  void Copy(const TMem& Mem) {

```

Comment:
Can we move multiline functions to .cpp file?

Reply:
Done for the more complex ones. 2-liners are still in the .h file